opentelemetry/instrumentations/distributor-node.ts

```diff
@@ -5,15 +5,21 @@ import { FsInstrumentation } from '@opentelemetry/instrumentation-fs'
 import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
 import { PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics'
 import { NodeSDK } from '@opentelemetry/sdk-node'
-import { BatchSpanProcessor } from '@opentelemetry/sdk-trace-node'
+import { BatchSpanProcessor, Span } from '@opentelemetry/sdk-trace-node'
 import { ClientRequest, ServerResponse } from 'http'
 
 /** Opentelemetry Instrumentation for Joystream Distributor Node */
 
+class CustomSpanProcessor extends BatchSpanProcessor {
+  onStart(span: Span) {
+    span.setAttribute('nodeId', process.env.JOYSTREAM_DISTRIBUTOR__ID)
```

Comment:
Question: I see the code in the config parser setting this env variable, but it happens in the distributor code. Do we know that `onStart()` is called after the config parser has done it's job?

Reply:
Good question, the instrument code is loaded & initialized first and then the config parser will do its job, i.e. exporting the env vars. Because of this reason the first few spans when the distributor node starts will have nodeId as undefined.